src/adapter/src/coord.rs

```diff
@@ -788,10 +783,39 @@ impl<S: Append + 'static> Coordinator<S> {
         }
     }
 
-    #[allow(dead_code)]
-    async fn create_storage_export(&mut self, id: GlobalId, sink: &Sink) {
+    async fn create_storage_export(
+        &mut self,
+        id: GlobalId,
+        sink: &Sink,
+        connection: StorageSinkConnection,
+    ) -> Result<(), AdapterError> {
+        // Validate `sink.from` is in fact a storage collection
+        self.controller.storage().collection(sink.from)?;
+
+        // The AsOf is used to determine at what time to snapshot reading from the persist collection.  This is
+        // primarily relevant when we do _not_ want to include the snapshot in the sink.  Choosing now will mean
+        // that only things going forward are exported.
+        let timeline = self
+            .get_timeline(sink.from)
+            .unwrap_or(Timeline::EpochMilliseconds);
+        let now = if timeline == Timeline::EpochMilliseconds {
+            let now = self.now();
+            now.step_back().unwrap_or(now)
+        } else {
+            // For non realtime sources, we define now as the largest timestamp, not in
+            // advance of any object's upper. This is the largest timestamp that is closed
+            // to writes.
+            let id_bundle = self.ids_in_timeline(&timeline);
+            self.largest_not_in_advance_of_upper(&id_bundle)
+        };
```

Comment:
This I think should be replaced with something like `self.ensure_timeline_state(timeline).await.oracle.read_ts()`. The logic here is used to advance timelines which includes additional logic to make sure that these number never go backwards. Instead you want to use the current timestamp on your timeline.